blogService/src/main/java/com/blog/blogservice/service/impl/ArticleServiceImpl.java

```diff
@@ -43,6 +46,9 @@ private boolean isUserAbleToCreatePost(final ArticleDto articleDto, final Long b
     @Override
     public ArticleDto getArticleById(final Long articleId) {
         Optional<Article> articleOptional = articleRepository.findById(articleId);
+        //poniżej dodałam Optionala dla commentarzy
+        Optional<Set<Comment>> commentsSetOptional = Optional.ofNullable(articleRepository.findById(articleId).get().getComments());
+
```

Comment:
magicznie komentarze się same nie dodadzą. tak jak już pare razy wspominałam, kolejny krok to zmapowanie tych komentarzy, czyli:
(uwzględniając ze optional set zamienisz po prostu na List<Comment>)
```
List<CommenDto> commentDtos = commentsFromDb.stream()
                  .map(comment -> commentMapper.map(comment, CommentDto.class))
                  .collect(Collectors.toList());
```
a następnie najlepiej dokonać troche zmian typu:
```
if (articleOptional.isPresent()) {
  ArticleDto articleDto = articleMapper.map(articleOptional.get(), ArticleDto.class);
  articleDto.setComments(comments);
  return articleDto;
}
return null;
```

Reply:
teraz zauważyłam po twoim ostatnim komentarzu. po cholere robisz ponownie articleRepository.findById(articleId, skoro masz już ten obiekt pobrany z bazy i trzymasz pod articleOptional. zamiast ponownie wywoływać zrób articleOptional.get().getComments()
Musisz pamiętać aby korzystać z danych które już są pobrane z bazy.

Reply:
zgadzam się w 100%